frontend/src/views/CellGuide/components/common/OntologyDagView/index.tsx

```diff
@@ -181,25 +187,68 @@ export default function OntologyDagView({
   // other properties like the positions of the nodes.
   const { data: rawTree } = useCellOntologyTree();
 
-  const entityId = cellTypeId ?? tissueId ?? "";
-  const useTreeState = cellTypeId
-    ? useCellOntologyTreeState
-    : useCellOntologyTreeStateTissue;
-  // Contains the nodes that are initially expanded (expandedNodes) and the nodes that are collapsed
-  // by default when their parents are expanded (notShownWhenExpanded).
-  // Also optionally contains cell counts for each node in the tissue ontology view (tissueCounts).
-  const { data: initialTreeState } = useTreeState(entityId);
+  const { data: initialTreeStateCell } = useCellOntologyTreeState(
+    cellTypeId ?? ""
+  );
+  const { data: initialTreeStateTissue } = useCellOntologyTreeStateTissue(
+    tissueId ?? ""
+  );
+  const initialTreeState: InitialCellOntologyTreeStateResponse | undefined =
```

Comment:
This now handles the case where the initial tree state for both cell type and tissue is defined. We take the intersection of isExpanded and union of notShownWhenExpanded.

Reply:
Maybe worth adding the comment in the code!